### PR TITLE
Registration part2

### PR DESCRIPTION
**Pull request by rishikaharithaMSIT**
Please review my code

### CODE REVIEW

File: application.py [hunk 1]
```diff
@@ -52,4 +52,10 @@ def register():
...
             responseMessage = "You registered sucessfully!"
         return render_template("index.html",data=responseMessage)
-    return render_template("register.html")
\ No newline at end of file
+    return render_template("register.html")
+
+@app.route("/admin")
+def listUsers():
+    queryResultSet = Users.query.all()
+    # print(queryResultSet[0])
```
Comment (praveengarimella) [inline application.py]: clean up the commented code

File: templates/admin.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+<head>
+  <title>Admin</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css">
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.16.0/umd/popper.min.js"></script>
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"></script>
+</head>
+<body>
```
Comment (praveengarimella) [inline templates/admin.html]: How about moving the common HTML that is used across the website into a common template?

File: application.py [hunk 3]
```diff
@@ -52,4 +52,10 @@ def register():
             db.session.commit()
             responseMessage = "You registered sucessfully!"
         return render_template("index.html",data=responseMessage)
-    return render_template("register.html")
\ No newline at end of file
+    return render_template("register.html")
+
+@app.route("/admin")
+def listUsers():
+    queryResultSet = Users.query.all()
```
Comment (praveengarimella) [inline application.py]: Does this display the list in the descending order of the user creation time?

File: application.py [hunk 4]
```diff
@@ -12,14 +14,18 @@
     raise RuntimeError("DATABASE_URL is not set")
 
 # Configure session to use filesystem
-app.config["SESSION_PERMANENT"] = False
-app.config["SESSION_TYPE"] = "filesystem"
-Session(app)
+# app.config["SESSION_PERMANENT"] = False
+# app.config["SESSION_TYPE"] = "filesystem"
```
Comment (praveengarimella) [inline application.py]: remove commented code

File: application.py [hunk 5]
```diff
@@ -30,7 +36,20 @@ def index():
 def register():
     if request.method == "POST":
         username = request.form['username']
+        email = request.form['email']
         password = request.form['pswd']
-        print(username, password)
```
Comment (praveengarimella) [inline application.py:34]: use logging module instead of print statements for debugging. the advantage is that you can turn on the debug statements when needed.

File: application.py [hunk 6]
```diff
@@ -30,7 +36,20 @@ def index():
...
-        return render_template("index.html",data=username)
+        registerTime = datetime.now()
```
Comment (praveengarimella) [inline application.py]: this can be moved to the constructor of the user object.

File: application.py [hunk 7]
```diff
@@ -30,7 +36,20 @@ def index():
...
+        print(username,email,registerTime,password)
+
+        queryResultSet = Users.query.filter_by(email=email).all()
+
+        responseMessage = ""
+        if(len(queryResultSet) > 0):
+            responseMessage = "The email id you have used already exists!"
+        else:
```
Comment (praveengarimella) [inline application.py]: Simplyfy this by catching an exception at the time of inserting the record. SQLAlchemy throws the IntegrityError.

**Review by praveengarimella — CHANGES_REQUESTED** [review]
Please review and fix the issues that I've raised.

File: application.py (hunk 3 above)
Comment (rishikaharithaMSIT) [reply application.py]: No. I changed it to : 
`queryResultSet = Users.query.order_by(desc(Users.registerTime)).all()`

**Review by praveengarimella — COMMENTED** [review]
great 👍
